src/common/d3Components/tooltip/tooltip.ts

```diff
@@ -0,0 +1,58 @@
+import { D3Selection } from '../../types';
+import { cnc } from '../../helpers/classname';
+import { mouse } from 'd3';
+const styles = require('./tooltip.scss');
+
+interface Props {
+  node: D3Selection;
+  xOffset?: number;
+  yOffset?: number;
+}
+
+interface State {
+  tooltip: D3Selection;
+}
+
+export const tooltipComponent = (props: Props) => {
+  const state = createEmptyState();
+  state.tooltip = render(props);
```

Comment:
We are taking here concepts from react and applying them to d3 components isn't it?